lib_camera/src/isp/camera_isp_eof_auto_exposure.c

```diff
@@ -0,0 +1,264 @@
+
+    for (int k = 0; k < HIST_BIN_COUNT; k++) {
+        uint32_t bin_count = histogram->bins[k];
+        // mean
+        temp_mean += bin_count * k;
+        // max and min
+        if (bin_count != 0) {  // the last that is not zero
+            temp_max = k;
+            if (temp_min == 0) { // first time is zero, then min is set
+                temp_min = k;
+            }
+        }
+    }
+    // max and min count
+    stats->max_count = histogram->bins[temp_max];
+    stats->min_count = histogram->bins[temp_min];
+
+    // biased downwards due to truncation
+    stats->max = (temp_max << HIST_QUANT_BITS);
+    stats->min = (temp_min << HIST_QUANT_BITS);
+    stats->mean = (temp_mean) * (1 << HIST_QUANT_BITS) * inv_img_size;
+}
+
+static
+void stats_skewness(
+    channel_histogram_t* histogram,
+    channel_stats_t* stats,
+    const float inv_img_size)
+{
+    const float zk_values[] = {
+      -1.000000, -0.907753, -0.821362, -0.740633, -0.665375, -0.595396,
+      -0.530504, -0.470508, -0.415214, -0.364431, -0.317968, -0.275632,
+      -0.237231, -0.202574, -0.171468, -0.143721, -0.119142, -0.097538,
+      -0.078717, -0.062488, -0.048659, -0.037037, -0.027431, -0.019648,
+      -0.013497, -0.008786, -0.005323, -0.002915, -0.001372, -0.000005,
+      -0.000108,    -4e-06,     4e-06,  0.000108,    0.0005,  0.001372,
+       0.002915,  0.005323,  0.008786,  0.013497,  0.019648,  0.027431,
+       0.037037,  0.048659,  0.062488,  0.078717,  0.097538,  0.119142,
+       0.143721,  0.171468,  0.202574,  0.237231,  0.275632,  0.317968,
+       0.364431,  0.415214,  0.470508,  0.530504,  0.595396,  0.665375,
+       0.740633,  0.821362,  0.907753,  1.0 };
+
+    float skew = 0.0;
+    for (int k = 0; k < HIST_BIN_COUNT; k++) {
+        skew += zk_values[k] * histogram->bins[k];
+    }
+
+    // Normnalization
+    stats->skewness = skew * inv_img_size;
+}
+
+static inline
+void stats_compute_stats(
+    statistics_t* stats,
+    histograms_t* histograms,
+    const float inv_img_size)
+{
+    stats_simple(&histograms->histogram_red, &stats->stats_red, inv_img_size);
+    stats_simple(&histograms->histogram_green, &stats->stats_green, inv_img_size);
+    stats_simple(&histograms->histogram_blue, &stats->stats_blue, inv_img_size);
+
+    stats_skewness(&histograms->histogram_red, &stats->stats_red, inv_img_size);
+    stats_skewness(&histograms->histogram_green, &stats->stats_green, inv_img_size);
+    stats_skewness(&histograms->histogram_blue, &stats->stats_blue, inv_img_size);
+}
+
+static inline
+float stats_compute_mean_skewness(statistics_t* stats)
+{
+    float mean = \
+        stats->stats_red.skewness + \
+        stats->stats_green.skewness + \
+        stats->stats_blue.skewness;
+    mean /= 3.0;
+    return mean;
+}
+
+static inline
+int8_t csign(float x)
+{
+    return (x > 0) - (x < 0);
+}
+
+static
+uint8_t AE_compute_new_exposure(float exposure, float skewness)
+{
+    static float a = 1;     // minimum value for exposure

```

Comment:
It would not surprise me if the compiler generates a warning at the int to float conversion required by these variable definitions (and the assignments on line 172).  If it does, then I suggest that you initialize each variable with a floating point constant since excess warnings make it difficult for anyone reading the compiler's output messages to see important information.